src/sentry/snuba/metrics/extraction.py

```diff
@@ -905,7 +905,30 @@ def _deep_sorted(value: Union[Any, Dict[Any, Any]]) -> Union[Any, Dict[Any, Any]
 
 
 def are_specs_equal(spec_1: MetricSpec, spec_2: MetricSpec) -> bool:
-    return _deep_sorted(spec_1) == _deep_sorted(spec_2)
```

Comment:
In the previous commits I was making changes to `_deep_sorted` but it was not turning out pretty in any way.